### ISSUE TITLE
Default view shows transactions created at the current date instead of the transactions of the day

### ISSUE THREAD

**Issue by Braayy** [NickvisionApps/Denaro#746]
# Explanation
Hi! I don't know if this is intended, but when i create a transaction with a different date than today, the default view still shows the just created transaction. I think the reason is because the view queries all transaction with the creation date or transaction date equal to today, instead of just the ones with the transaction date equal to today.

# Example
Here i create a transaction with date 2023/12/04.
![Captura de tela de 2023-12-02 22-15-58](https://github.com/NickvisionApps/Denaro/assets/28508164/12070025-fc3a-46c4-90fd-6e39599b6c8a)

And the view, shows the just created transaction, even if the transaction date is 2023/12/04. The day when i took the screenshot was 2023/12/02.
![Captura de tela de 2023-12-02 22-16-36](https://github.com/NickvisionApps/Denaro/assets/28508164/ab1da0ff-57bc-48f5-b8f5-ecea030230be)

I then created another transaction with the today date in it, just to show that my assumption, that the view shows all transactions with creation date or transaction date equal to today, is true.
![Captura de tela de 2023-12-02 22-17-06](https://github.com/NickvisionApps/Denaro/assets/28508164/45209a50-6e19-44ec-ae56-679b450f0aea)

If i click on the calendar on 2023/12/04, it only shows the Testing transaction and not Testing 2 transaction.

# Conclusion
If this is intended, it could be helpful to add an configuration to disable this behavior, as it is a little annoying for me at least.

If this isn't intended, i could try to create a pr for solving this issue.

Thanks in advance

**Comment by JoseBritto** [comment]
Thanks for the detailed report! Currently Denaro treats selecting current date as if no date is selected. This isn't ideal, and I'm not sure how to go about fixing this. Currently @nlogozzo is a bit busy, he is the one who can tackle this issue. 


My suggestion is maybe have the Today button be called "Reset" or something similar, which shows all transactions without any date filter and select the current month on the calendar view instead of the current day. I'm not sure how to do this, I tried calling this [method](https://docs.gtk.org/gtk3/method.Calendar.select_month.html) but got stuck as I don't know how to do to convert Int32 to guint.  After we have this then we can treat selecting the current day as any other date and by default on app startup we can have no date selected and show all transactions. This should hopefully solve this problem.

**Comment by nlogozzo** [comment]
> Currently Denaro treats selecting current date as if no date is selected. This isn't ideal, and I'm not sure how to go about fixing this. 

The reason for this is because the Calendar has no way of "unselecting" a date, that is a date is always selected in the Calendar. Therefore, we made the decision when we first made the app to have today's date be kind of a "Today" overview of all of your transactions in your account and then when you select a different date in the calendar it will actually filter the transactions on those dates.

One thing you could do is sort by decreasing day to have today's transactions always show up at the top of the list.

But like @JoseBritto said, I don't know of a better design for this.

> I'm not sure how to do this, I tried calling this [method](https://docs.gtk.org/gtk3/method.Calendar.select_month.html) but got stuck as I don't know how to do to convert Int32 to guint.

The problem with this method is it will change the selected month, but the day will still be highlighted/selected since like I said above, the calendar cannot have no dates selected.

**Comment by JoseBritto** [comment]
> The problem with this method is it will change the selected month, but the day will still be highlighted/selected since like I said above, the calendar cannot have no dates selected.

Ah. I didn't know that

**Comment by nlogozzo** [comment]
Yeah so I don't really know how to solve this without a rework of the ui/controls used.

**Comment by nlogozzo** [comment]
The C++ rewrite is bringing a brand new design to the entire application. More to follow as we develop the new version :)
